src/Symfony/Bridge/Doctrine/Listener/DoctrineConnectionListener.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Listener;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception as DBALException;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+use ProxyManager\Proxy\LazyLoadingInterface;
+use Symfony\Bridge\Doctrine\Event\ForceKernelRebootEvent;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\VarExporter\LazyObjectInterface;
+
+/**
+ * Based on https://github.com/Baldinof/roadrunner-bundle/blob/3.x/src/Integration/Doctrine/DoctrineORMMiddleware.php.
+ */
+class DoctrineConnectionListener implements EventSubscriberInterface
+{
+    public function __construct(private ManagerRegistry $managerRegistry, private ContainerInterface $container, private EventDispatcherInterface $eventDispatcher)
+    {
+    }
+
+    public function onKernelRequest(RequestEvent $event)
+    {
+        $connectionServices = $this->managerRegistry->getConnectionNames();
+
+        foreach ($connectionServices as $connectionServiceName) {
+            if (!$this->container->initialized($connectionServiceName)) {
+                continue;
+            }
+
+            $connection = $this->container->get($connectionServiceName);
+
+            if (!$connection instanceof Connection) {
+                throw new \RuntimeException(sprintf('Is not an instance of "%s".', Connection::class));
```

Comment:
I think this doesn't match usual messages format. I think the codebase tends to have something more like "Connection is not an instance of...", same for the others